t/unit/app/test_preload_cli.py

```diff
@@ -0,0 +1,55 @@
+import sys
+from click.testing import CliRunner
+from celery.bin.celery import celery
+
+
+def test_preload_options(isolated_cli_runner: CliRunner):
+    # Verify commands like shell and purge can accept preload options.
+    # Projects like Pyramid-Celery's ini option should be valid preload
+    # options.
+
+    res_without_preload = isolated_cli_runner.invoke(
+        celery,
+        ["-A", "t.unit.bin.proj.app", "purge", "-f", "--ini", "some_ini.ini"],
+        catch_exceptions=True,
+    )
+
+    assert "No such option: --ini" in res_without_preload.stdout
+    assert res_without_preload.exit_code == 2
+
+    res_without_preload = isolated_cli_runner.invoke(
+        celery,
+        ["-A", "t.unit.bin.proj.app", "shell", "--ini", "some_ini.ini"],
+        catch_exceptions=True,
+    )
+
+    assert "No such option: --ini" in res_without_preload.stdout
+    assert res_without_preload.exit_code == 2
+
+    res_with_preload = isolated_cli_runner.invoke(
+        celery,
+        [
+            "-A",
+            "t.unit.bin.proj.pyramid_celery_app",
+            "purge",
+            "-f",
+            "--ini",
+            "some_ini.ini",
+        ],
+        catch_exceptions=True,
+    )
+
```

Comment:
Is there a recommended way to fully clean/reset the applications called by the isolated_cli_runner? I noticed that if I run the `t.unit.bin.proj.pyramid_celery_app`  cases first then the --ini option seems to be defined even when running the `t.unit.bin.proj.app`. I'm guessing that Celery has a global or some other persistence that I am not aware of that needs to get reset or added to the tear-down of  the tests. I've also tried running these as separate tests and even in separate test files and there still seems to be some persistence or leakage of the user_options. Basically I want two separate and independent Celery processes to test and configure here.

Reply:
I left a TODO in the code with the issue laid out. I'm fine leaving as it is since I believe it will address the underlying options support.